### ISSUE TITLE
bug in arp_helper

### ISSUE THREAD

**Issue by nemethf** [noxrepo/pox#202]
[This recent commit ](https://github.com/noxrepo/pox/commit/d8bbb33343d524b80bdb14a83823eafc3bbf615d) breaks arp_helper
because  the `use_port_mac` argument has been removed form ARPHelper.__init__(), but 
at [line  237](https://github.com/noxrepo/pox/blob/d8bbb33343d524b80bdb14a83823eafc3bbf615d/pox/proto/arp_helper.py#L237) the corresponding member variable / attribute is still referenced.

Thanks

**Comment by sjtutyx** [comment]
Thanks for reminding. We will fix it later.

**Comment by MurphyMc** [comment]
Hah, well, the commit message warns it may break code, but I don't think that was what I meant. ;)  It looks like I just forgot to finish it.  I don't have time to test at the moment, but [here's a quick pass at fixing it](https://gist.github.com/MurphyMc/602e6da045bbc72e3104c404cdadd833).  If it seems to work for you, I'll just push it.  Otherwise, I can take another crack at it later.

**Comment by nemethf** [comment]
Yes, that code fixes the problem for our very simple test-case.  Thank you.

**Comment by MurphyMc** [comment]
Fixed in eel and fangtooth.  Thanks for the report and test!

**Comment by nemethf** [comment]
Thanks again.

(Does fangtooth mean that you happened to find time to work on pox once again?  That would be really great news.)

**Comment by MurphyMc** [comment]
Yeah, POX has been factoring into work again, so it has been getting some attention.  My focus hasn't been OpenFlow type stuff, but even that has been getting some work.  RIP-on-OVS, OVS bundle action, improvements to discovery, a new and much-improved OpenFlow spanning tree component (spanning_forrest), and the little discovery graph, for example.  But I think the mass of stuff is probably not directly OpenFlow-related... GRE, VXLAN, IGMPv3 support in the packet library... config files... lots of improvements/additions to the DHCP and ARP stuff in pox/proto/... a bunch of stuff for working with Linux network interfaces including a revent-based capture API... lots of little fixes and improvements here and there.  (Indeed, some of that stuff is actually about making it useful in non-OpenFlow contexts, e.g., the DHCP client code.)

It's still living in my (public) fork, but I'm planning to shift it over to noxrepo one of these days.

In the slightly longer term, I'm planning to finally mainline the [ovsdb](https://github.com/MurphyMc/pox/tree/experimental/pox/ovsdb) support I wrote and the [edge apps](https://github.com/garyachy/pox/tree/edge/pox/edge) that @garyachy and I worked on a few years ago (oof), though that probably all needs to be tested against modern OVS first, so it might be a while before I get to it.
